lib/agenda/stop.js

```diff
@@ -25,6 +25,12 @@ module.exports = function() {
         return resolve();
       }
 
+      debug('clearing out job lock expiration timers');
+      self._lockExpirationTimers.forEach(timer => {
```

Comment:
The `for-of` loop is better in all circumstances IMO. More readable. Better CPU and RAM performance.
```js
for (const timer of self._lockExpirationTimers) clearTimeout(timer);
self._lockExpirationTimers.clear();
```